src/units.cpp

```diff
@@ -265,6 +268,103 @@ bool Units::isBaseUnit() const
     return unitCount() == 0 && standardUnitCheck;
 }
 
+ptrdiff_t ulpsDistance(const double a, const double b)
+{
+    const auto max = std::numeric_limits<ptrdiff_t>::max();
+
+    // Max distance for NaN
+    if (std::isnan(a) || std::isnan(b)) {
+        return max;
+    }
+
+    // If one's infinite and they're not equal, max distance.
+    if ((std::isinf(a) || std::isinf(b)) && (std::isinf(a) != std::isinf(b))) {
+        return max;
+    }
+
+    ptrdiff_t ia;
+    ptrdiff_t ib;
+    memcpy(&ia, &a, sizeof(double));
+    memcpy(&ib, &b, sizeof(double));
+
+    // Return the absolute value of the distance in ULPs.
+    ptrdiff_t distance = ia - ib;
+    if (distance < 0) {
+        distance = -distance;
+    }
+
+    return distance;
+}
+
+/**
+ * @brief Decide if two doubles are nearly equal.
```

Comment:
Should this have @c double and @c float?

Reply:
I don't think so, it is not saying it is a double like the param documentation.